### PR TITLE
Add swiftlint support #127

### LINKED ISSUE #127: Swift support
**Possible linters**

- [swiftlint](https://github.com/realm/SwiftLint) tool is widely using across swift community. 
- [swiftformat](https://github.com/nicklockwood/SwiftFormat) is also popular code formatting tool that have --lint option.

### PR DESCRIPTION
**Pull request by engali94**
Add linting support for Swift language using __swiftlint__ 

Fixes #127

### CODE REVIEW

File: lib/worker.sh [hunk 1]
```diff
@@ -37,7 +37,7 @@ function LintCodebase() {
   #####################################
   # Validate we have linter installed #
   #####################################
-  VALIDATE_INSTALL_CMD=$(command -v "${LINTER_NAME}" 2>&1)
+  VALIDATE_INSTALL_CMD=$(command -v "${LINTER_NAME}" 2>&1 || (command version "${LINTER_NAME}" 2>&1) )
```
Comment (admiralAwkbar) [inline lib/worker.sh:40]: nice touch

File: lib/linter.sh [hunk 2]
```diff
@@ -197,7 +197,8 @@ VALIDATE_PYTHON_FLAKE8="${VALIDATE_PYTHON_FLAKE8}"                   # Boolean t
 VALIDATE_RAKU="${VALIDATE_RAKU}"                                     # Boolean to validate language
 VALIDATE_RUBY="${VALIDATE_RUBY}"                                     # Boolean to validate language
 VALIDATE_STATES="${VALIDATE_STATES}"                                 # Boolean to validate language
-VALIDATE_SQL="${VALIDATE_SQL}"                                       # Boolean to validate language
+VALIDATE_SQL="${VALIDATE_SQL}"
```
Comment (admiralAwkbar) [inline lib/linter.sh:200]: fix the spacing and comment?

**Comment by admiralAwkbar** [comment]
@engali94 really nice!
I'll see what help I can add to this as well :)

File: lib/linter.sh (hunk 2 above)
Comment (engali94) [reply lib/linter.sh:200]: ohh, @admiralAwkbar  thanks I will fix it

**Comment by admiralAwkbar** [comment]
@engali94 Can you add to the `docs/disabling-linters.md` the information from https://github.com/realm/SwiftLint#disable-rules-in-code

That should be super helpful

**Comment by admiralAwkbar** [comment]
@engali94 seems the download of the swift binary is not going well... may need to troubleshoot that

**Comment by engali94** [comment]
> @engali94 Can you add to the `docs/disabling-linters.md` the information from https://github.com/realm/SwiftLint#disable-rules-in-code
> 
> That should be super helpful

Yes, it Will be added thanks...

**Comment by admiralAwkbar** [comment]
@engali94 interesting error...
```bash
2020-08-07 16:54:54 [INFO  ]   File:[/tmp/lint/.automation/automation/swift/swift_good_1.swift]
2020-08-07 16:54:54 [ERROR ]   Found errors in [swiftlint] linter!
2020-08-07 16:54:54 [ERROR ]   [swiftlint: error while loading shared libraries: libswiftCore.so: cannot open shared object file: No such file or directory]
2020-08-07 16:54:54 [ERROR ]   Linter CMD:[swiftlint /tmp/lint/.automation/automation/swift/swift_good_1.swift]
```

**Comment by engali94** [comment]
@admiralAwkbar
Another interesting error, can you help with it please?
```bash
2020-08-07 21:07:47 [ERROR ]   Failed to find [ansible-lint] in system!
2020-08-07 21:07:47 [FATAL ]   []
```

**Comment by admiralAwkbar** [comment]
So fixed all the issues, but the major issue:
Swiftlint was built on Mac and doesn't have a great way to get it into Alpine... Maybe we compile and build it?

**Comment by engali94** [comment]
> So fixed all the issues, but the major issue:
> Swiftlint was built on Mac and doesn't have a great way to get it into Alpine... Maybe we compile and build it?

@admiralAwkbar Thanks for fixing the issue, The problem with swiftlint is it needs Swift to be installed to run properly as shown in [Dockerfile](https://hub.docker.com/r/norionomura/swiftlint/dockerfile) the package is built as a static executable which means it should work on Linux.

File: Dockerfile [hunk 3]
```diff
@@ -297,13 +305,15 @@ ENV ACTIONS_RUNNER_DEBUG=${ACTIONS_RUNNER_DEBUG} \
...
+    VALIDATE_SWIFT=${VALIDATE_SWIFT} \
     VALIDATE_TERRAFORM=${VALIDATE_TERRAFORM} \
     VALIDATE_TERRAFORM_TERRASCAN=${VALIDATE_TERRAFORM_TERRASCAN} \
     VALIDATE_TYPESCRIPT_ES=${VALIDATE_TYPESCRIPT_ES} \
     VALIDATE_TYPESCRIPT_STANDARD=${VALIDATE_TYPESCRIPT_STANDARD} \
     VALIDATE_XML=${VALIDATE_XML} \
-    VALIDATE_YAML=${VALIDATE_YAML}
-
+    VALIDATE_YAML=${VALIDATE_YAML}\
+    LD_LIBRARY_PATH=/usr/lib/swift/linux
```
Comment (GaboFDC) [inline Dockerfile:315]: Please move this var to keep the block sorted

**Pull request closed without merging**